### PR TITLE
refactor(ThemeScanner): Twig tokenization and tests

### PR DESCRIPTION
**Pull request by matteotrubini**
Refactor `ThemeScanner::processStandardTags()` to reuse Twig tokenization for more robust extraction instead of regex-based parsing. Introduce a statically cached collection of registered plugin filters instead of using hardcoded values.

Move all related tests into a new `ThemeScannerTest` class and expand coverage to include edge cases, Unicode input, and special characters.

## Summary by CodeRabbit

* **Refactor**
  * Reworked translation scanning to a token-based parser for more reliable extraction.
* **Chores**
  * Added caching for translatable filter names to improve performance.
  * Removed an obsolete test tied to the prior scanner.
* **Tests**
  * Added comprehensive tests covering null/empty input handling, parsing accuracy, caching behavior, and diverse tag patterns.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: classes/ThemeScanner.php [hunk 1]
```diff
@@ -137,43 +142,76 @@ public function scanMailTemplatesForMessages()
...
+        $translatableStrings = [];
+
+        // Fetch registered filters once for performance (cached statically)
+        if (self::$translatableFilters === null) {
+            self::$translatableFilters = array_keys(
+                \System\Classes\PluginManager::instance()
+                    ->findByIdentifier('Winter.Translate')
+                    ->registerMarkupTags()['filters']
+            );
+        }
```
[the remark this answers is not in the document]
Comment (matteotrubini) [reply classes/ThemeScanner.php]: @mjauvin any thoughts on whether we should assume the plugin is registered here, making a fatal error acceptable?

[the remark this answers is not in the document]
Comment (LukeTowers) [reply classes/ThemeScanner.php]: @coderabbitai how on earth would this class even be loaded if that plugin didn't exist in the first place? Bad rabbit.

**Comment by LukeTowers** [comment]
@matteotrubini in the future if you could [enable edits by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests) that would make it easier for me to review and merge.

**Comment by matteotrubini** [comment]
@mjauvin PHP 7.4 is no longer relevant: the plugin now requires PHP >= 8.1, and the CI has been updated accordingly.
